contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+                    amount1Min: 0,
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    // needs to be 0 because the pool is already created
+                    // non zero amount attempts to create a new instance of the pool
+                    sqrtPriceX96: 0
+                })
+            );
+
+            tokenId = mintedTokenId;
+
+            updateUnderlyingAssets();
+            emit LiquidityAdded(
+                WETHBalance, // WETHAmountDesired
+                OETHbRequired, // OETHbAmountDesired
+                WETHAmountSupplied, // WETHAmountSupplied
+                OETHbAmountSupplied, // OETHbAmountSupplied
+                mintedTokenId, // tokenId
+                underlyingAssets
+            );
+        } else {
+
+            (,uint256 WETHAmountSupplied, uint256 OETHbAmountSupplied) = positionManager.increaseLiquidity(
+                INonfungiblePositionManager.IncreaseLiquidityParams({
+                    tokenId: tokenId,
+                    amount0Desired: WETHBalance,
+                    amount1Desired: OETHbRequired,
+                    // we don't need slippage protection checking for that in _checkLiquidityWithinExpectedShare
+                    amount0Min: 0,
+                    // we don't need slippage protection checking for that in _checkLiquidityWithinExpectedShare
+                    amount1Min: 0,
+                    deadline: block.timestamp
+                })
+            );
+
+            updateUnderlyingAssets();
+            emit LiquidityAdded(
+                WETHBalance, // WETHAmountDesired
+                OETHbRequired, // OETHbAmountDesired
+                WETHAmountSupplied, // WETHAmountSupplied
+                OETHbAmountSupplied, // OETHbAmountSupplied
+                tokenId, // tokenId
+                underlyingAssets
+            );
+        }
+
+        // burn remaining OETHb
+        _burnOETHbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    /**
+     * @dev Check that the liquidity in the pool is withing the expected WETH to OETHb ratio
+     */
+    function _checkLiquidityWithinExpectedShare() internal {
+        (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+        require(WETHPositionBalance + OETHbPositionBalance > 0, "Can not withdraw full position");
+
+        uint160 currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(currentPrice >= sqrtRatioX96Tick0 && currentPrice <= sqrtRatioX96Tick1, "Not in expected tick range");
+
+        uint256 currentWethShare = 0;
+        if (WETHPositionBalance != 0) {
+            currentWethShare = WETHPositionBalance.divPrecisely(WETHPositionBalance + OETHbPositionBalance);
+        }
+
+        if (currentWethShare < poolWethShareVarianceAllowed || // uint256 mustn't go below 0  
```

Comment:
🔴 This check: `currentWethShare < poolWethShareVarianceAllowed` does not feel right. Not really things on both sides that should be compared with each other.

Let's say admins set a 30% target +- 20%. A `currentWethShare` of 15% would be both inside what the admins intended, and a perfectly computable value, but would revert here.

Reply:
It's also something of a pattern we'd like to avoid to have config values that can successfully be set to values, which then later cause surprise reverts when the code is used. I'd lean toward making the requires here work with any valid config value, rather than having them revert on certain combinations.

Reply:
Thanks great point. I'm not sure what I was thinking about overcomplicating the code so much. It can be simplified so nicely: 
https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/f61034b181fd42bd0bf139e3e39142f5f5958a3b